lib/src/components/NxAccordion/stateful/__tests__/NxStatefulAccordion.test.tsx

```diff
@@ -4,46 +4,255 @@
+            onToggle = jest.fn(),
+            el = quickRender({
+              onToggle,
+              children: (
+                <NxAccordion.Header>
+                  <NxAccordion.Title>Foo</NxAccordion.Title>
+                </NxAccordion.Header>
+              )});
+        const header = el.getByRole('button')!;
+        expect(onToggle).not.toHaveBeenCalled();
+
+        await user.click(header);
+
+        expect(onToggle).toHaveBeenCalled();
+      });
+
+      it('toggles the accordion open prop', async function() {
+        const user = userEvent.setup(),
+            onToggle = jest.fn(),
+            el = quickRender({
+              onToggle,
+              children: (
+                <NxAccordion.Header>
+                  <NxAccordion.Title>Foo</NxAccordion.Title>
+                </NxAccordion.Header>
+              )});
+        const header = el.getByRole('button')!;
+        expect(el.getByRole('group')).not.toHaveAttribute('open');
+
+        await user.click(header);
+
+        expect(el.getByRole('group')).toHaveAttribute('open');
+      });
+    });
+
+    describe('when the accordion is currently open', function() {
+      it('calls onToggle', async function() {
+        const user = userEvent.setup();
+        const onToggle = jest.fn();
+
+        const { container } = quickRender({
+          onToggle,
+          defaultOpen: true,
+          children: (
+            <NxAccordion.Header>
+              <span>Foo</span>
+            </NxAccordion.Header>
+          )});
+
+        const header = container.querySelector<HTMLElement>('summary')!;
+
+        expect(onToggle).not.toHaveBeenCalled();
+
+        await user.click(header);
+
+        expect(onToggle).toHaveBeenCalledTimes(1);
+
+        await user.click(container);
+
+        expect(onToggle).toHaveBeenCalledTimes(1);
+      });
+
+      it('toggles the accordion open prop', async function() {
+        const user = userEvent.setup();
+        const onToggle = jest.fn();
+
+        const el = quickRender({
+          onToggle,
+          defaultOpen: true,
+          children: (
+            <NxAccordion.Header>
+              <span>Foo</span>
+            </NxAccordion.Header>
+          )});
+
+        const header = el.getByRole('button')!;
+
+        expect(el.getByRole('group')).toHaveAttribute('open');
+
+        await user.click(header);
+
+        expect(el.getByRole('group')).not.toHaveAttribute('open');
+      });
+    });
+
+    describe('when a non-button element in the header with its own onClick handler is clicked', function() {
+      it('changes the open state and fires onToggle', async function() {
+        const user = userEvent.setup();
+        const titleOnClick = jest.fn(),
+            onToggle = jest.fn();
+
+        const { container } = render(
+          <NxStatefulAccordion onToggle={onToggle}>
+            <NxAccordion.Header>
+              <NxAccordion.Title onClick={titleOnClick}>Foo</NxAccordion.Title>
```

Comment:
This relies on the fact that `NxAccordion.Title` can actually take an `onClick` prop, which isn't really the point.  I think it'd be cleaner to wrap the "Foo" in a span with a testid and put the `onClick` handler on that

Reply:
> This relies on the fact that `NxAccordion.Title` can actually take an `onClick` prop, which isn't really the point. I think it'd be cleaner to wrap the "Foo" in a span with a testid and put the `onClick` handler on that

Makes sense, updated here: [141fd74](https://github.com/sonatype/sonatype-react-shared-components/pull/879/commits/141fd742930b5eadea1da2665c234b418138d1cb)